### ISSUE TITLE
Errors in logs: handler svccontroller: empty namespace

### ISSUE THREAD

**Issue by hmonsalv** [rancher/rke2#3463]
**Environmental Info:**
RKE2 Version: 
```
rke2 version v1.23.10+rke2r1 (2616002259455211fb8af2e2eb8f996593c18b86)
go version go1.17.5b7
```
Rancher Version:  `2.6.8`

Node(s) CPU architecture, OS, and Version: 
`Linux ip-10-99-0-30 5.15.0-1019-aws #23~20.04.1-Ubuntu SMP Thu Aug 18 03:20:14 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux`

Cluster Configuration:
Single server

**Describe the bug:**
The log is getting flooded with errors syncing due to  handler svccontroller.

**Steps To Reproduce:**
- Create a new RKE2 cluster in Rancher:
   -  Select k8s version: `v1.23.10+rke2r1`
   - Unselect NGINX Ingress
   - Edit yaml: select `cni: none`
- On the server, create manifest folder and copy the cilium-install.yaml: `/var/lib/rancher/rke2/server/manifests/cilium-install.yaml`
- Execute the provided bootstrap command on the server.
- Check rke2-server logs: `journalctl -u rke2-server -f`

**Expected behavior:**
Cluster is functional with no errors

**Actual behavior:**
Cluster is functional with errors:
```
Oct 14 11:13:47 ip-10-99-0-30 rke2[2637]: time="2022-10-14T11:13:47Z" level=error msg="error syncing 'kube-system/rke2-coredns-rke2-coredns': handler svccontroller: an empty namespace may not be set when a resource name is provided, requeuing"
Oct 14 11:13:49 ip-10-99-0-30 rke2[2637]: time="2022-10-14T11:13:49Z" level=error msg="error syncing 'kube-system/hubble-enterprise': handler svccontroller: an empty namespace may not be set when a resource name is provided, requeuing"
Oct 14 11:13:49 ip-10-99-0-30 rke2[2637]: time="2022-10-14T11:13:49Z" level=error msg="error syncing 'kube-system/hubble-relay': handler svccontroller: an empty namespace may not be set when a resource name is provided, requeuing"
Oct 14 11:13:49 ip-10-99-0-30 rke2[2637]: time="2022-10-14T11:13:49Z" level=error msg="error syncing 'kube-system/hubble-peer': handler svccontroller: an empty namespace may not be set when a resource name is provided, requeuing"
Oct 14 11:13:49 ip-10-99-0-30 rke2[2637]: time="2022-10-14T11:13:49Z" level=error msg="error syncing 'cattle-system/cattle-cluster-agent': handler svccontroller: an empty namespace may not be set when a resource name is provided, requeuing"
Oct 14 11:14:05 ip-10-99-0-30 rke2[2637]: time="2022-10-14T11:14:05Z" level=error msg="error syncing 'monitoring/prometheus-k8s': handler svccontroller: an empty namespace may not be set when a resource name is provided, requeuing"
Oct 14 11:14:15 ip-10-99-0-30 rke2[2637]: time="2022-10-14T11:14:15Z" level=error msg="error syncing 'cert-manager/cert-manager-webhook': handler svccontroller: an empty namespace may not be set when a resource name is provided, requeuing"
Oct 14 11:14:17 ip-10-99-0-30 rke2[2637]: time="2022-10-14T11:14:17Z" level=error msg="error syncing 'kube-system/kubelet': handler svccontroller: an empty namespace may not be set when a resource name is provided, requeuing"
Oct 14 11:14:19 ip-10-99-0-30 rke2[2637]: time="2022-10-14T11:14:19Z" level=error msg="error syncing 'kube-system/hubble-ui': handler svccontroller: an empty namespace may not be set when a resource name is provided, requeuing"
```

**Additional context / logs:**
I saw that same issues were tackled here; https://github.com/rancher/rke2/issues/3254 for k8s version `v1.24.4-rc1+rke2r1`, but in that case were provoked by MetalLB. In our case we do not use MetalLB, and K8s version is different, so it might be due to any other issue.

I am not sure if this is relevant, but we don't use any of the default CNI provided by rke2, and instead we use Cilium Enterprise.

However, in order to discard this could be the issue, I have created a new RKE2 cluster using the default Cilium available (rke2-cilium) and the same errors appear. They always appear after rke2-server service starts all the controllers:

```
Oct 14 11:46:42 ip-10-99-0-204 rke2[2473]: I1014 11:46:42.479792    2473 request.go:665] Waited for 1.184975112s due to client-side throttling, not priority and fairness, request: PUT:https://127.0.0.1:6443/api/v1/nodes/ip-10-99-0-204
Oct 14 11:46:42 ip-10-99-0-204 rke2[2473]: time="2022-10-14T11:46:42Z" level=info msg="Event(v1.ObjectReference{Kind:\"Addon\", Namespace:\"kube-system\", Name:\"rke2-cilium\", UID:\"d759e72d-57db-443d-814a-7d59eac4a778\", APIVersion:\"k3s.cattle.io/v1\", ResourceVersion:\"427\", FieldPath:\"\"}): type: 'Normal' reason: 'ApplyingManifest' Applying manifest at \"/var/lib/rancher/rke2/server/manifests/rke2-cilium.yaml\""
Oct 14 11:46:42 ip-10-99-0-204 rke2[2473]: time="2022-10-14T11:46:42Z" level=info msg="Event(v1.ObjectReference{Kind:\"Addon\", Namespace:\"kube-system\", Name:\"rke2-cilium\", UID:\"d759e72d-57db-443d-814a-7d59eac4a778\", APIVersion:\"k3s.cattle.io/v1\", ResourceVersion:\"427\", FieldPath:\"\"}): type: 'Normal' reason: 'AppliedManifest' Applied manifest at \"/var/lib/rancher/rke2/server/manifests/rke2-cilium.yaml\""
Oct 14 11:46:42 ip-10-99-0-204 rke2[2473]: time="2022-10-14T11:46:42Z" level=info msg="Event(v1.ObjectReference{Kind:\"HelmChart\", Namespace:\"kube-system\", Name:\"rke2-cilium\", UID:\"27de47cd-d92f-4fa6-aa3f-c078d4c67af2\", APIVersion:\"helm.cattle.io/v1\", ResourceVersion:\"428\", FieldPath:\"\"}): type: 'Normal' reason: 'ApplyJob' Applying HelmChart using Job kube-system/helm-install-rke2-cilium"
Oct 14 11:46:42 ip-10-99-0-204 rke2[2473]: time="2022-10-14T11:46:42Z" level=info msg="Event(v1.ObjectReference{Kind:\"HelmChart\", Namespace:\"kube-system\", Name:\"rke2-cilium\", UID:\"27de47cd-d92f-4fa6-aa3f-c078d4c67af2\", APIVersion:\"helm.cattle.io/v1\", ResourceVersion:\"438\", FieldPath:\"\"}): type: 'Normal' reason: 'ApplyJob' Applying HelmChart using Job kube-system/helm-install-rke2-cilium"
Oct 14 11:46:43 ip-10-99-0-204 rke2[2473]: time="2022-10-14T11:46:43Z" level=info msg="Event(v1.ObjectReference{Kind:\"Addon\", Namespace:\"kube-system\", Name:\"rke2-coredns\", UID:\"e2d1919f-a1fa-44c8-96d4-fe4ddb8eb456\", APIVersion:\"k3s.cattle.io/v1\", ResourceVersion:\"442\", FieldPath:\"\"}): type: 'Normal' reason: 'ApplyingManifest' Applying manifest at \"/var/lib/rancher/rke2/server/manifests/rke2-coredns.yaml\""
Oct 14 11:46:43 ip-10-99-0-204 rke2[2473]: time="2022-10-14T11:46:43Z" level=info msg="Event(v1.ObjectReference{Kind:\"Addon\", Namespace:\"kube-system\", Name:\"rke2-coredns\", UID:\"e2d1919f-a1fa-44c8-96d4-fe4ddb8eb456\", APIVersion:\"k3s.cattle.io/v1\", ResourceVersion:\"442\", FieldPath:\"\"}): type: 'Normal' reason: 'AppliedManifest' Applied manifest at \"/var/lib/rancher/rke2/server/manifests/rke2-coredns.yaml\""
Oct 14 11:46:43 ip-10-99-0-204 rke2[2473]: time="2022-10-14T11:46:43Z" level=info msg="Event(v1.ObjectReference{Kind:\"HelmChart\", Namespace:\"kube-system\", Name:\"rke2-coredns\", UID:\"c6fec8bc-f77a-4d3b-81a7-359028e457b6\", APIVersion:\"helm.cattle.io/v1\", ResourceVersion:\"443\", FieldPath:\"\"}): type: 'Normal' reason: 'ApplyJob' Applying HelmChart using Job kube-system/helm-install-rke2-coredns"
Oct 14 11:46:43 ip-10-99-0-204 rke2[2473]: time="2022-10-14T11:46:43Z" level=info msg="Event(v1.ObjectReference{Kind:\"HelmChart\", Namespace:\"kube-system\", Name:\"rke2-coredns\", UID:\"c6fec8bc-f77a-4d3b-81a7-359028e457b6\", APIVersion:\"helm.cattle.io/v1\", ResourceVersion:\"452\", FieldPath:\"\"}): type: 'Normal' reason: 'ApplyJob' Applying HelmChart using Job kube-system/helm-install-rke2-coredns"
Oct 14 11:46:43 ip-10-99-0-204 rke2[2473]: time="2022-10-14T11:46:43Z" level=info msg="Event(v1.ObjectReference{Kind:\"Addon\", Namespace:\"kube-system\", Name:\"rke2-ingress-nginx\", UID:\"\", APIVersion:\"k3s.cattle.io/v1\", ResourceVersion:\"\", FieldPath:\"\"}): type: 'Normal' reason: 'DeletingManifest' Deleting manifest at \"/var/lib/rancher/rke2/server/manifests/rke2-ingress-nginx.yaml\""
Oct 14 11:46:43 ip-10-99-0-204 rke2[2473]: time="2022-10-14T11:46:43Z" level=info msg="Starting /v1, Kind=Endpoints controller"
Oct 14 11:46:43 ip-10-99-0-204 rke2[2473]: time="2022-10-14T11:46:43Z" level=info msg="Starting /v1, Kind=Secret controller"
Oct 14 11:46:43 ip-10-99-0-204 rke2[2473]: time="2022-10-14T11:46:43Z" level=info msg="Starting /v1, Kind=ConfigMap controller"
Oct 14 11:46:43 ip-10-99-0-204 rke2[2473]: time="2022-10-14T11:46:43Z" level=info msg="Starting /v1, Kind=Service controller"
Oct 14 11:46:43 ip-10-99-0-204 rke2[2473]: time="2022-10-14T11:46:43Z" level=info msg="Starting /v1, Kind=Pod controller"
Oct 14 11:46:43 ip-10-99-0-204 rke2[2473]: time="2022-10-14T11:46:43Z" level=info msg="Starting /v1, Kind=ServiceAccount controller"
Oct 14 11:46:43 ip-10-99-0-204 rke2[2473]: time="2022-10-14T11:46:43Z" level=error msg="error syncing 'cattle-system/cattle-cluster-agent': handler svccontroller: an empty namespace may not be set when a resource name is provided, requeuing"
Oct 14 11:46:43 ip-10-99-0-204 rke2[2473]: time="2022-10-14T11:46:43Z" level=error msg="error syncing 'default/kubernetes': handler svccontroller: an empty namespace may not be set when a resource name is provided, requeuing"
Oct 14 11:46:43 ip-10-99-0-204 rke2[2473]: time="2022-10-14T11:46:43Z" level=error msg="error syncing 'cattle-system/cattle-cluster-agent': handler svccontroller: an empty namespace may not be set when a resource name is provided, requeuing"
Oct 14 11:46:43 ip-10-99-0-204 rke2[2473]: time="2022-10-14T11:46:43Z" level=error msg="error syncing 'default/kubernetes': handler svccontroller: an empty namespace may not be set when a resource name is provided, requeuing"
Oct 14 11:46:43 ip-10-99-0-204 rke2[2473]: time="2022-10-14T11:46:43Z" level=error msg="error syncing 'default/kubernetes': handler svccontroller: an empty namespace may not be set when a resource name is provided, requeuing"
Oct 14 11:46:43 ip-10-99-0-204 rke2[2473]: time="2022-10-14T11:46:43Z" level=error msg="error syncing 'cattle-system/cattle-cluster-agent': handler svccontroller: an empty namespace may not be set when a resource name is provided, requeuing"
Oct 14 11:46:43 ip-10-99-0-204 rke2[2473]: time="2022-10-14T11:46:43Z" level=error msg="error syncing 'cattle-system/cattle-cluster-agent': handler svccontroller: an empty namespace may not be set when a resource name is provided, requeuing"
Oct 14 11:46:43 ip-10-99-0-204 rke2[2473]: time="2022-10-14T11:46:43Z" level=error msg="error syncing 'default/kubernetes': handler svccontroller: an empty namespace may not be set when a resource name is provided, requeuing"
Oct 14 11:46:43 ip-10-99-0-204 rke2[2473]: time="2022-10-14T11:46:43Z" level=error msg="error syncing 'cattle-system/cattle-cluster-agent': handler svccontroller: an empty namespace may not be set when a resource name is provided, requeuing"
Oct 14 11:46:43 ip-10-99-0-204 rke2[2473]: time="2022-10-14T11:46:43Z" level=error msg="error syncing 'default/kubernetes': handler svccontroller: an empty namespace may not be set when a resource name is provided, requeuing"
Oct 14 11:46:43 ip-10-99-0-204 rke2[2473]: time="2022-10-14T11:46:43Z" level=error msg="error syncing 'cattle-system/cattle-cluster-agent': handler svccontroller: an empty namespace may not be set when a resource name is provided, requeuing"
```

**Comment by brandond** [comment]
This regression is fixed in the currently in-progess 1.23 patch release. While noisy, the error shouldn't affect the functionality of the cluster.

**Comment by hmonsalv** [comment]
Great! Thanks @brandond
